torch/distributed/tensor/parallel/_data_parallel_utils.py

```diff
@@ -1,24 +1,34 @@
 from typing import Optional, Tuple
 
 import torch
-from torch.distributed._tensor import DTensor as DistributedTensor
+from torch.distributed._functional_collectives import AsyncCollectiveTensor
+from torch.distributed._tensor import DTensor
 from torch.distributed._tensor.placement_types import DTensorSpec
 
 
+def sync_grad_hook(grad):
+    if isinstance(grad, AsyncCollectiveTensor):
+        grad.wait()
```

Comment:
@wanchaol Why do we discard the return value of `grad.wait()`, which unwraps the ACT? I.e., why not `grad = grad.wait()`?

Reply:
Good catch! I think we should unwrap yes, it's a bug. Could u help submit a fix on this?

Reply:
Oh wait, this looks like it was fixed on main already.